lib/active_merchant/billing/gateways/globalcollect.rb

```diff
@@ -0,0 +1,294 @@
+          "name" => options[:company]
+        }
+        post["order"]["contactDetails"] = {
+          "emailAddress" => options[:email],
+          "phoneNumber" => options[:phone]
+        }
+      end
+
+      def add_refund_customer_data(post, options)
+        if address = options[:billing_address] || options[:address]
+          post["customer"]["address"] = {
+            "countryCode" => address[:country]
+          }
+          post["customer"]["address"]["name"] = {
+            "firstName" => options[:firstname],
+            "surname" => options[:lastname]
+          }
+          post["customer"]["contactDetails"] = {
+            "emailAddress" => options[:email]
+          }
+        end
+      end
+
+      def add_address(post, creditcard, options)
+        billing_address = options[:billing_address] || options[:address]
+        shipping_address = options[:shipping_address]
+        if billing_address = options[:billing_address] || options[:address]
+          post["order"]["customer"]["billingAddress"] = {
+            "street" => billing_address[:address1],
+            "additionalInfo" => billing_address[:address2],
+            "zip" => billing_address[:zip],
+            "city" => billing_address[:city],
+            "state" => billing_address[:state],
+            "countryCode" => billing_address[:country]
+          }
+        end
+        if shipping_address
+          post["order"]["customer"]["shippingAddress"] = {
+            "street" => shipping_address[:address1],
+            "additionalInfo" => shipping_address[:address2],
+            "zip" => shipping_address[:zip],
+            "city" => shipping_address[:city],
+            "state" => shipping_address[:state],
+            "countryCode" => shipping_address[:country]
+          }
+          post["order"]["customer"]["shippingAddress"]["name"] = {
+            "firstName" => shipping_address[:firstname],
+            "surname" => shipping_address[:lastname]
+          }
+        end
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def url(action, authorization)
+        (test? ? test_url : live_url) + uri(action, authorization)
+      end
+
+      def uri(action, authorization)
+        uri = "/v1/#{@options[:merchant_id]}/"
+        case action
+        when :test_connection
+          uri + "services/testconnection"
+        when :authorize
+          uri + "payments"
+        when :capture
+          uri + "payments/#{authorization}/approve"
+        when :refund
+          uri + "payments/#{authorization}/refund"
+        when :void
+          uri + "payments/#{authorization}/cancel"
+        end
+      end
+
+      def commit(action, post, authorization = nil)
+        begin
+        response = parse(ssl_post(url(action, authorization), post.to_json, headers(action, post, authorization)))
+        rescue ResponseError => e
+          raise unless(e.response.code.to_s =~ /4\d\d/)
```

Comment:
Many other adapters use the regex there, but now that you point it out, this should in fact be 
`raise e if e.response.code.to_i >= 300`